ArduCopter/ekf_check.cpp

```diff
@@ -160,6 +136,17 @@ void Copter::failsafe_ekf_event()
     failsafe.ekf = true;
     Log_Write_Error(ERROR_SUBSYSTEM_FAILSAFE_EKFINAV, ERROR_CODE_FAILSAFE_OCCURRED);
 
+    // does this mode require position?
+    if (!copter.flightmode->requires_GPS() && (g.fs_ekf_action != FS_EKF_ACTION_LAND_EVEN_STABILIZE)) {
```

Comment:
hummm that is not right as it in LAND we don't require GPS so the EKF failsafe will do nothing, but we could be in GPS landing or preclanding, and this could lead to a crash. the check for landing should be the first IMO

Reply:
yes, you're absolutely right, thanks!  fixed in follow-up commit.